File: PTB.Core/Ledger/Ledger.cs
namespace PTB.Core.Ledger
{
    public struct Ledger
    {
        public int Index;
        public string Date, Amount, Subject, Title, Subcategory;
        public char Type, Locked;

        public Ledger(int index, string date, string amount, string subject, string title, char type, char locked, string subcategory)
        {
            Index = index;
            Date = date;
            Amount = amount;
            Subject = subject;
            Title = title;
            Type = type;
            Subcategory = subcategory;
            Locked = locked;
        }
    }
}

File: PTB.Web/ClientApp/src/app/ledger/ledger.ts
export interface ILedgerEntry {
  "index": number,
  "date": ILedgerColumn,
  "type": ILedgerColumn,
  "amount": ILedgerColumn,
  "subcategory": ILedgerColumn,
  "title": ILedgerColumn,
  "subject": ILedgerColumn,
  "locked": ILedgerColumn,
}

export interface ILedgerColumn {
  "name": string,
  "value": string,
  "index": number,
  "size": number,
  "offset": number,
  "editable": boolean
}

export interface IRow {
  "index": number,
  "columns": IColumn[]
}

export interface IColumn {
  "columnValue": string,
  "columnName": string,
  "index": number,
  "size": number,
  "offset": number,
  "editable": boolean
}